bigquery_etl/cli/query.py

```diff
@@ -918,24 +941,68 @@ def _run_query(
         if "query" not in query_arguments:
             query_arguments = ["query"] + query_arguments
 
+        query_text = render_template(
+            query_file.name,
+            template_folder=str(query_file.parent),
+            templates_dir="",
+            format=False,
+            **addl_templates,
+        )
+
+        # create a session, setting default project and dataset
+        if billing_project is not None:
+            if dataset_id is not None:
+                default_dataset = dataset_id
+            else:
+                _, default_dataset, _ = extract_from_query_path(query_file)
+
+            session_id = create_query_session(project_id, default_dataset)
+            query_arguments.append(f"--session_id={session_id}")
+
         # write rendered query to a temporary file;
         # query string cannot be passed directly to bq as SQL comments will be interpreted as CLI arguments
         with tempfile.NamedTemporaryFile(mode="w+") as query_stream:
-            query_stream.write(
-                render_template(
-                    query_file.name,
-                    template_folder=str(query_file.parent),
-                    templates_dir="",
-                    format=False,
-                    **addl_templates,
-                )
-            )
+            query_stream.write(query_text)
             query_stream.seek(0)
 
             # run the query as shell command so that passed parameters can be used as is
             subprocess.check_call(["bq"] + query_arguments, stdin=query_stream)
 
 
+def create_query_session(
+    project_id: Optional[str] = None, dataset_id: Optional[str] = None
+):
+    """Create a bigquery session and return the session id.
+
+    Optionally set the system variables @@dataset_project_id and @@dataset_id
+    if project_id and dataset_id are given. This sets the default project_id or dataset_id
+    for table/view/udf references that do not have a project or dataset qualifier.
+    """
+    query_parts = []
+    if project_id is not None:
+        query_parts.append(f"SET @@dataset_project_id = '{project_id}';")
+    if dataset_id is not None:
+        query_parts.append(f"SET @@dataset_id = '{dataset_id}';")
+
+    if len(query_parts) == 0:  # need to run a non-empty query
+        session_query = "SELECT 1"
+    else:
+        session_query = "\n".join(query_parts)
+
+    client = bigquery.Client()
+
+    job_config = bigquery.QueryJobConfig(
+        create_session=True,
+        use_legacy_sql=False,
+    )
+    result = client.query(session_query, job_config)
+
+    if result.session_info is None:
+        raise RuntimeError(f"Failed to get session id with {query}")
```

Comment:
I don't see a `query` variable.
```suggestion
        raise RuntimeError(f"Failed to get session id with {session_query}")
```